src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -280,15 +320,77 @@ private boolean parseNodes(List<Node> nodes) {
           continue;
         }
 
-        if (cpeUriCheck(cpeMatch)) {
+        if (projectCheck(cpeMatch, referenceMatch, project)) {
           return true;
         }
       }
 
-      if (parseNodes(node.getChildren())) {
+      if (parseNodes(node.getChildren(), referenceMatch, project)) {
         return true;
       }
     }
     return false;
   }
+
+  /**
+   * Checks if the referenceUrl of type {@link URI} matches with the project's URL.
+   * 
+   * @param referenceUrl of type {@link URI}.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the input URL matches the project's URL. Otherwise false.
+   */
+  private static boolean checkUrl(URI referenceUrl, GitHubProject project) {
+    return checkUrlHost(referenceUrl.getHost(), project)
+        && checkUrlPath(referenceUrl.getPath(), project);
+  }
+
+  /**
+   * Checks if the input host matches with project's URL host.
+   * 
+   * @param host The host of the URL.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the input host matches the project's URL host. Otherwise false.
+   */
+  private static boolean checkUrlHost(String host, GitHubProject project) {
+    return host.equals(project.url().getHost());
+  }
+
+  /**
+   * Matches the reference URL with the project's URL. The URL must not contain the pre-defined stop
+   * words.
+   * 
+   * @param path The path of the URL.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the project URL matches with the reference URL. Otherwise False.
+   */
+  private static boolean checkUrlPath(String path, GitHubProject project) {
+    String delimiter = "/";
+    return path != null
+        && checkSplitPath(StringUtils.stripStart(path, delimiter).split(delimiter), project);
+  }
+
+  /**
+   * Matches the split contents of the URL path. Each predefined position of the split path must
+   * match with its respective counterpart. The path must not contain the pre-defined stop words.
+   * 
+   * @param path The split path of the URL.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the reference path matches with the project path. Otherwise False.
+   */
+  private static boolean checkSplitPath(String[] path, GitHubProject project) {
+    return path.length > 3 
```

Comment:
Should it be `paht.length > 2`? Could you please add a test for the `checkUrlPath` method that covers all the cases we've discussed (and others)?

Reply:
👍